packages/material-ui/src/TextField/TextField.d.ts

```diff
@@ -36,6 +36,7 @@ export interface BaseTextFieldProps
   required?: boolean;
   rows?: string | number;
   rowsMax?: string | number;
+  size?: 'small' | 'medium';
```

Comment:
Question: would it be a good idea to make enums or types for string values as 'small' or 'medium' and then make a size combined type of `Size.Small | Size.medium`? I see many places use those inline values, which is a very minor thing but it leaves an opening for small typos

Reply:
@sarpt I have wondered about it too. I don't think that it's really important. I'm not aware of having faced issues with it in the past.
However, having outdated TypeScript definitions has been a common theme in the past. A migration to TypeScript would prevent us to forget about it #15984. To keep thinking about it. At the same time, only 10% of our users use the TypeScript version of the demos in our documentation, maybe it's not that important either.